### PR TITLE
ISLE: Add proper bool type

### PR DESCRIPTION
**Pull request by Kmeakin**
Partially solves https://github.com/bytecodealliance/wasmtime/issues/3573

### CODE REVIEW

File: cranelift/isle/isle/src/sema.rs [hunk 1]
```diff
@@ -80,9 +80,36 @@ pub struct TypeEnv {
     pub errors: Vec<Error>,
 }
 
+/// A built-in type.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum BuiltinType {
```
Comment (cfallin) [inline cranelift/isle/isle/src/sema.rs:85]: I'm not sure I understand the reason for having a notion of "built-in types" for bool but not for other types that we have constants for (namely integers) -- while on the other hand it adds some complication to the IR. Maybe we could separate out this part into another PR and discuss separately?

Comment (cfallin) [reply cranelift/isle/isle/src/sema.rs:85]: (To make it more explicit: I think we probably *should* have this eventually; but for integers too, and then we should check that integer literals are used only in contexts that expect them.)

**Review by cfallin — COMMENTED** [review]
Thanks for working on this -- it's nice to see motion on the "nice-to-have" upgrades we've had in our backlog for a while!

Mostly looks fine; my main thought is about the `BuiltinType` abstraction and whether we can separate the change out.

File: cranelift/isle/veri/README.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
 # Crocus: An SMT-based ISLE verification tool
 
-This directory contains Crocus, a tool for verifying instruction lowering and transformation rules written in ISLE. Crocus uses an underlying SMT solver to model values in ISLE rules in as logical bitvectors, searching over all possible inputs to find potential soundness counterexamples. The motivation and context project are described in detail in our ASPLOS 2024 paper: [Lightweight, Modular Verification for WebAssembly-to-Native Instruction Selection](https://dl.acm.org/doi/10.1145/3617232.3624862).
```
[the remark this answers is not in the document]
Comment (Kmeakin) [reply cranelift/isle/veri/README.md:3]: I'm not sure how this slipped in, I don't recall editing it. Anyway, I'll revert it

File: cranelift/isle/isle/src/sema.rs (hunk 1 above)
Comment (Kmeakin) [reply cranelift/isle/isle/src/sema.rs:85]: The intention is to extend `BuiltinType` to include other types (like `u{8,16,32,64,size}, `i{8,16,32,64,size}` or `str`) in future PRs

Comment (cfallin) [reply cranelift/isle/isle/src/sema.rs:85]: OK, yeah, on further thought I suppose it's totally fine to migrate integers over as a second step -- as long as the intention is there to finish out the refactor. Thanks!

**Review by cfallin — APPROVED** [review]
